incubator/hnc/pkg/reconcilers/hnc_config_test.go

```diff
@@ -212,6 +213,25 @@ var _ = Describe("HNCConfiguration", func() {
 		Expect(hasObject(ctx, "Secret", barName, "foo-sec-2")()).Should(BeFalse())
 
 	})
+
+	It("should reconcile after adding a new crd to the apiserver", func() {
+		// Add a config for a type that hasn't been defined yet.
+		addToHNCConfig(ctx, "stable.example.com/v1", "CronTab", api.Propagate)
+
+		// The corresponding object reconciler should not be created because the type does not exist.
+		Eventually(hasHNCConfigurationConditionWithMsg(ctx, api.ObjectReconcilerCreationFailed, "stable.example.com/v1, Kind=CronTab")).Should(BeTrue())
+
+		// Add the CRD for CronTab to the apiserver.
+		createCronTabCRD(ctx)
+		setParent(ctx, barName, fooName)
+		makeObject(ctx, "CronTab", fooName, "foo-crontab")
+
```

Comment:
Is there a reason you're also calling `setParent` and `makeObject` _before_ checking that the condition has been cleared?

Reply:
I was trying to group "actions" together then put "expectations" together. We can also do the following:

```go
	
 // Add the CRD for CronTab to the apiserver.
createCronTabCRD(ctx)

// The object reconciler for CronTab should be created successfully.
Eventually(hasHNCConfigurationConditionWithMsg(ctx, api.ObjectReconcilerCreationFailed, "stable.example.com/v1, Kind=CronTab")).Should(BeFalse())

setParent(ctx, barName, fooName)
makeObject(ctx, "CronTab", fooName, "foo-crontab")

// "foo-crontab" should be propagated from foo to bar.
Eventually(hasObject(ctx, "CronTab", barName, "foo-crontab")).Should(BeTrue())
Expect(objectInheritedFrom(ctx, "CronTab", barName, "foo-crontab")).Should(Equal(fooName))
```
Actually I think this is more clear. I will change to that for now. Feel free to let me know if there are any suggestions. :)